src/library_webgl.js

```diff
@@ -2168,7 +2173,22 @@ var LibraryGL = {
     }
 #endif
 
-    GLctx.uniform1iv(GL.uniforms[location], {{{ makeHEAPView('32', 'value', 'value+count*4') }}});
+#if GL_POOL_TEMP_BUFFERS
+    if (count <= GL.MINI_TEMP_BUFFER_SIZE) {
+      // avoid allocation when uploading few enough uniforms
+      var view = GL.miniTempBufferIntViews[count-1];
+      for (var i = 0; i < count; ++i) {
+        view[i] = {{{ makeGetValue('value', '4*i', 'i32') }}};
+      }
```

Comment:
Could TypedArray.set() work here? It seems like the miniTempBufferViews are correctly sized so this could work.

Reply:
> Could TypedArray.set() work here? It seems like the miniTempBufferViews are correctly sized so this could work.

It would, and I think in ancient times we did have a `.set()` call here, but it was profiled to be slower. (on UE4) Generally these functions are called with some 1-4 element long arrays, and the loop+assignment is done already in the same time as the prelude that JS engines do on fetching and validating the source and target typedarrays that are being set.